src/components/Pagination/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { type FC, useState, useEffect, ReactNode } from 'react';
+import { Box, Button, HStack, Icon } from '@chakra-ui/react';
+import {
+  ArrowLeftIcon,
+  ArrowRightIcon,
+  ChevronLeftIcon,
+  ChevronRightIcon,
+} from '@chakra-ui/icons';
+
+type PropTypes = {
+  size?: 'md' | 'sm' | 'xs';
+  pageIndex: number;
+  limit: number;
+  count: number;
+  setPageIndex: (index: number) => void;
+};
+export const Pagination: FC<PropTypes> = ({
+  count,
+  limit,
+  pageIndex,
+  setPageIndex,
+  size = 'sm',
+}) => {
+  const [items, setItems] = useState<ReactNode[]>([]);
+  const [sumPages] = useState(Math.ceil(count / limit));
+
+  useEffect(() => {
+    if (pageIndex >= 0) {
```

Comment:
Đừng làm kiểu này, những thứ tính toán phức tạp và render jsx không nên dùng trong useEffect.